### PR TITLE
[FEATURE] Adds Query Include/Exclude Package

### PR DESCRIPTION
**Pull request by jdamner**
Change Details
-------------
A project suggested it would be great to have the ability to specify which posts are included or excluded from a query-loop block. This package aims to provide this feature. The functionality already exists in core, but there's no interface to control it. At the moment this package provides a simple interface.

Further enhancements could be acheievd in the future, such as limiting the number of selectable posts to include to the same limit as the query and only returning posts that meet other filters selected in the query for the include/exclude.

Also, there's potential performance issues if the number of posts or pages is excessive so we could look at more lazily loading these.

Screenshots
------------
Screenshot 2024-07-12 at 16 35 31

### CODE REVIEW

File: packages/query-include-exclude/src/SelectPostsControl.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			tokens.map( ( token ) => {
+				if ( 'string' === typeof token ) {
+					const maybeNumber = parseInt( token, 10 );
+					if ( ! isNaN( maybeNumber ) ) {
+						return maybeNumber;
+					}
+
+					return (
+						allPosts.find(
+							( post ) => post.title.rendered === token
```
Comment (jdamner) [inline packages/query-include-exclude/src/SelectPostsControl.tsx:52]: This is my least favourite line. The FormTokenField component will only return the string-value and can't return an object with ID + Label. Here we're matching the title from the array of all posts to determine the ID. Seems a little flaky IMO, but makes this implementation much easier by using `FormTokenField` than trying to implement that interface ourselves.

File: packages/query-include-exclude/query-include-exclude.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Plugin Name: QueryIncludeExclude
+ * Description: A brief description of the plugin.
```
Comment (Ritchbox) [inline packages/query-include-exclude/query-include-exclude.php:4]: :D

File: packages/query-include-exclude/src/SelectPostsControl.tsx (hunk 1 above)
Comment (jdamner) [reply packages/query-include-exclude/src/SelectPostsControl.tsx]: Turns out WP core team face the same issue and rely on the same technique. Not super happy with that, but in the context of terms/taxonomies duplicate titles are far less likely than for posts/pages. Have updated the code to include the post ID in the selector which has double benefits of both ensuring that the post ID is always accurate, and now you can search by ID 🎉

**Review by Ritchbox — APPROVED** [review]
LGTM

**Pull request merged**